File: src/main/java/net/guizhanss/fastmachines/utils/Keys.java
package net.guizhanss.fastmachines.utils;

import javax.annotation.Nonnull;

import com.google.common.base.Preconditions;

import org.bukkit.NamespacedKey;

import net.guizhanss.fastmachines.FastMachines;

import lombok.experimental.UtilityClass;

@SuppressWarnings("ConstantConditions")
@UtilityClass
public final class Keys {
    public static NamespacedKey get(@Nonnull String name) {
        Preconditions.checkArgument(name != null, "name cannot be null");

        return new NamespacedKey(FastMachines.getInstance(), name);
    }
}

File: src/main/kotlin/net/guizhanss/fastmachines/utils/constants/Keys.kt
package net.guizhanss.fastmachines.utils.constants

import net.guizhanss.fastmachines.FastMachines
import org.bukkit.NamespacedKey

object Keys {

    val MAIN_GROUP = fmKey("fast_machines")
    val MATERIALS = fmKey("materials")
    val MACHINES = fmKey("machines")
    val HIDDEN = fmKey("hidden")
    val DISPLAY_ITEM = fmKey("display_item")
}

internal fun fmKey(key: String) = NamespacedKey(FastMachines.instance, key)
